cli-plugins/manager/hooks.go

```diff
@@ -0,0 +1,61 @@
+package manager
+
+import (
+	"encoding/json"
+
+	"github.com/docker/cli/cli-plugins/hooks"
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+)
+
+// HookPluginData is the type representing the information
+// that plugins declaring support for hooks get passed when
+// being invoked following a CLI command execution.
+type HookPluginData struct {
+	RootCmd string
+}
+
+// RunPluginHooks calls the hook subcommand for all present
+// CLI plugins that declare support for hooks in their metadata
+// and parses/prints their responses.
+func RunPluginHooks(dockerCli command.Cli, rootCmd, subCommand *cobra.Command) error {
+	plugins, err := ListPlugins(dockerCli, rootCmd)
+	if err != nil {
+		return err
+	}
+	subCmdName := subCommand.Name()
```

Comment:
This line seems to cause a panic for command like `docker -v`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x104da7788]

goroutine 1 [running]:
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Name(0x105649148?)
	/Users/tonistiigi/gocode/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:1497 +0x18
github.com/docker/cli/cli-plugins/manager.RunPluginHooks({0x105649148, 0x140001b1400}, 0x0?, 0x14000463310?)
	/Users/tonistiigi/gocode/src/github.com/docker/cli/cli-plugins/manager/hooks.go:26 +0x44
main.runDocker(0x140001b1400)
	/Users/tonistiigi/gocode/src/github.com/docker/cli/cmd/docker/docker.go:336 +0x44c
main.main()
	/Users/tonistiigi/gocode/src/github.com/docker/cli/cmd/docker/docker.go:35 +0x98
```

Reply:
Thanks! I'll TAL